packages/dds/map/src/directory.ts

```diff
@@ -895,16 +951,23 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
     }
 
     public dispose(error?: Error): void {
-        this._disposed = true;
+        this._deleted = true;
         this.emit("disposed", this);
```

Comment:
Either we should avoid emitting this event until the subdirectory can't possibly be undisposed (op is ack'ed), or there should be some corresponding undispose event.

Reply:
Opened internal bug 1029 to track cleaning this up.